feature/visited/src/main/kotlin/uk/govuk/app/visited/data/VisitedItemsTransformer.kt

```diff
@@ -0,0 +1,78 @@
+package uk.govuk.app.visited.data
+
+import uk.govuk.app.visited.data.model.VisitedItem
+import uk.govuk.app.visited.ui.model.VisitedUi
+import java.time.LocalDate
+
+class VisitedItemsTransformer(
+    private val visitedItems: List<VisitedItem>,
+    private val todaysDate: LocalDate
+) {
+    val todaysItems: List<VisitedItem>
+    val thisMonthsItems: List<VisitedItem>
+    val previousMonthsItems: List<VisitedItem>
+    val groupedPreviousMonthsItems: Map<String, List<VisitedItem>>
+
+    init {
+        todaysItems = todaysItems()
+        thisMonthsItems = thisMonthsItems()
+        previousMonthsItems = previousMonthsItems()
+        groupedPreviousMonthsItems = groupPreviousMonthsItems()
+    }
+
+    fun transform(): Map<String, List<VisitedUi>> {
+        var groupedVisitedItems = emptyMap<String, List<VisitedUi>>()
```

Comment:
This should be a mutable map as concatenation of immutable map creates lots of redundant objects.